src/language/class-modifiers.md

```diff
@@ -0,0 +1,240 @@
+---
+title: Class modifiers
+description: Modifier keywords for class declarations to control external library access.
+---
+
+Class modifiers control how a class or a mixin can be implemented, extended,
+or mixed in. 
+
+The full set of modifiers that can appear before a class declaration are `base`,
+`final`, `interface`, `sealed`, `abstract`, and `mixin`. 
+Only the `base` modifier can appear before a mixin declaration. The modifiers do
+not apply to other declarations like `enum`, `typedef`, or `extension`.
+
+When deciding whether or not to use class modifiers, consider the following questions:
+1. Is [any access](#no-modifier) fine?
+2. If not, what do you want to restrict?
+    - [Implementation](#base)?
+    - [Inheritance](#interface)?
+    - [Both](#final)?
+    - [Subtyping](#sealed) by outside libraries at all?
+3. Then, does it need to be [abstract](#abstract)?
+4. And lastly, should it be a [class, mixin, or mixin class](#class-mixin-or-mixin-class)?
+
+## No modifier
+
+Use a `class` or `mixin` declaration without a modifier to **allow unrestricted
+access** to outside libraries. The standard behaviors apply:
+
+- [Construct][] new instances of a class
+- [Extend][] a class to create a new subtype
+- [Implement][] a class's interface
+- [Mix in][] a mixin or mixin class
+
+## `base`
+
+Use the `base` modifier to **disallow implementation** of a class or mixin.
+A base class can be extended and constructed. This applies trasitively to all
+subtypes, because implementing a subtype implements the supertype as well.  
+
+{{site.alert.note}}
+  Only the `base` modifier can accompany a mixin declaration. 
+  - A `base mixin` can be mixed in, but not implemented.
+  - A `base mixin class` can be constructed, extended, and mixed in, but not
+  implemented.
+{{site.alert.end}}
+
+```dart
+base class Vehicle {
+  String make;
+  String model;
+  void moveForward(int meters) { … }
+}
+
+var myCar = Vehicle(     // Can be constructed
```

Comment:
The class as defined doesn't have a constructor with this signature.  Is this intentional?  It feels better to me if someone could copy this code into an IDE and observe the results themselves.

Reply:
> Is this intentional?

I just copied these examples from some presentation slides; is there a reason that it'd be intentionally written incorrectly? Maybe to prove some other point? I'm not sure. I wrote this in another reply, but would the correct way be:
```dart
interface class Vehicle {
  String make?;
  String model?;
  void moveForward(int meters) { … }
}
// ...
var myCar = Vehicle();        // Can be constructed
```

Reply:
To be honest, I don't think there's a lot of value in showing an example of construction.  Construction is controlled orthogonally - it has nothing to do with `interface` or `base` etc.  It's controlled by `abstract`.  So I personally would just leave the constructor part off.  But if you feel that it's important to call out that in the absence of `abstract` an `interface class` can be constructed, then yes, what you have now works.  Or you could just leave off the `make` and `model` fields - they don't add much.